### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by mconigliaro** [SublimeLinter/SublimeLinter3#227]
This happens when I enable rbenv (which loads the chefdk version of Ruby for me). This error prevents the linter from working properly:

```
SublimeLinter: ruby: default.rb ['/Users/mconigliaro/.rbenv/shims/ruby', '-wc'] 
SublimeLinter: ruby output:
/opt/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults/operating_system.rb:15: warning: method redefined; discarding old user_dir
/opt/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults.rb:75: warning: previous definition of user_dir was here
-:17: syntax error, unexpected end-of-input, expecting keyword_end 
SublimeLinter: error in SublimeLinter daemon: 
SublimeLinter: -------------------- 
SublimeLinter: Traceback (most recent call last):
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 58, in loop
    item = self.q.get(block=True, timeout=self.MIN_DELAY)
  File "./queue.py", line 175, in get
queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 66, in loop
    self.lint(view_id, timestamp)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 112, in lint
    self.callback(view_id, timestamp)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 123, in lint
    Linter.lint_view(view, filename, code, hit_time, callback)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 917, in lint_view
    linter.lint(hit_time)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 1387, in lint
    self.highlight.range(line, pos, length=0, error_type=error_type, word_re=self.word_re)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/highlight.py", line 234, in range
    start, end = self.full_line(line)
  File "/Users/mconigliaro/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/highlight.py", line 202, in full_line
    start = self.newlines[line] + char_offset
IndexError: list index out of range
 
SublimeLinter: -------------------- 
```

When I disable rbenv and use the system ruby, I don't get the error anymore, and everything works fine:

```
SublimeLinter: ruby: default.rb ['/usr/bin/ruby', '-wc'] 
SublimeLinter: ruby output:
-:17: syntax error, unexpected end-of-input, expecting keyword_end 
```

I believe the error is being caused by these warnings from chefdk's rubygems:

```
/opt/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults/operating_system.rb:15: warning: method redefined; discarding old user_dir
/opt/chefdk/embedded/lib/ruby/site_ruby/2.1.0/rubygems/defaults.rb:75: warning: previous definition of user_dir was here
```

**Comment by mconigliaro** [comment]
Looks like someone fixed this in chefdk: https://github.com/chef/chef-dk/issues/302

But I wonder if this is something you might want to handle in the linter too (e.g. ignore warnings/errors from files I'm not editing).

**Comment by jawshooah** [comment]
You should submit this issue on [SublimeLinter-ruby](https://github.com/SublimeLinter/SublimeLinter-ruby). You're right, this can be fixed at the linter level by overriding `Linter#split_match` and checking the filename.

**Comment by skj3gg** [comment]
@jawshooah fix has been released with SL-ruby version 1.0.17
